### ISSUE TITLE
Add support for schedules that only happen when someone is home

### ISSUE THREAD

**Issue by AndrewX192** [arcus-smart-home/arcusplatform#64]
There doesn't seem to be a good way to have an appliance turn on at a certain time if someone is home / the alarm is unlocked, or if a specific person is present. It would be nice to add this to the scheduling system, or add the ability to create a rule that fires at a given time if someone is home.

Potentially explore adding this to scheduler-service or rule-service. I looked into the latter in the past, but struggled because rules can't fire at a user controlled time.

**Comment by msilvermanks** [comment]
Yeah we tossed around the idea of something like this but of course never implemented it, would be a cool feature but a decent amount of work
